aws/utils/spin_lock.h

```diff
@@ -27,22 +30,81 @@ class TicketSpinLock : boost::noncopyable {
       : now_serving_(0),
         next_ticket_(0) {}
 
+#ifdef DEBUG
+  struct DebugStats {
+    std::uint64_t acquired_count;
+    std::uint64_t acquired_with_lock;
+    std::uint64_t total_spins;
+  };
+
+  void add_acquired();
+  void add_acquired_lock();
+  void add_spin();
+
+  DebugStats get_debug_stats();
+#else
+#define add_acquired();
+#define add_acquired_lock();
+#define add_spin();
+#endif
+
+  
   inline void lock() noexcept {
     size_t my_ticket = next_ticket_.fetch_add(1);
-    while (now_serving_ != my_ticket);
+    std::uint32_t spin_count = 0;
+    do {
+      add_spin();
+      spin_count++;
+      if (spin_count > kMaxSpinCount) {
+        std::unique_lock<std::mutex> lock(lock_for_ticket(my_ticket));
+        cv_for_ticket(my_ticket).wait(lock, [this, &my_ticket] { return now_serving_ == my_ticket; });
```

Comment:
could you add a comment here describing what the update means when the lock is acquired.